lib/tfpdf/wp-tfpdf.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * To avoid modifying tFPDF directly we introduce WP_tFPDF, 
+ * A thin wrapper around the parent class to utilize WP_Filesystem 
+ * 
+ * Parent's coding standards are preserved deliberately.
+ */
+// phpcs:disable WordPress.WP.AlternativeFunctions.file_system_read_fwrite, WordPress.WP.AlternativeFunctions.file_system_read_fread, WordPress.WP.AlternativeFunctions.file_system_read_fopen, WordPress.WP.AlternativeFunctions.file_system_read_fclose, WordPress.VIP.FileSystemWritesDisallow
+if ( ! defined( 'ABSPATH' ) ) exit; // Exit if accessed directly
+
+require_once ABSPATH . '/wp-admin/includes/file.php';
+
+class WP_tFPDF extends tFPDF {
+	function __construct( $orientation = 'P', $unit = 'mm', $size = 'A4' ) {
+		parent::__construct( $orientation, $unit, $size );
+		$this->init_wp_filesystem();
+	}
+	
+	function init_wp_filesystem() {
+		global $wp_filesystem;
+
+		if ( ! is_a( $wp_filesystem, 'WP_Filesystem_Base' ) ) {
+			$creds = request_filesystem_credentials( site_url() );
+			$fs = wp_filesystem( $creds );
+			return $fs ?: new WP_Error( 'fs-init-error', "Couldn't initialize Filesystem" );
```

Comment:
Maybe a regular `if` statement would be a little clearer here?

Also, we never actually use this return value. Maybe it would be better to throw an exception instead?

I'm also not sure we need to `return true` below.

Reply:
So the logic goes, if the $wp_filesystem is not initialized, try to do so. `wp_filesystem( $creds )` returns boolean TRUE/FALSE. If it's FALSE, return a WP_Error. 

In case the filesystem is initialized properly - just return TRUE.

As for Elvis operator instead of a regular `if/else` is a personal preference, I think it looks cleaner than nested ifs.

Reply:
Re: initialization - I've addressed this workflow in my latest review.

Re: Elvis operator - this will probably be going away anyway, but in case it doesn't, I think we should remove the use of the Elvis operator. I don't mind it personally, but it's unconventional as far as our coding standards go, and it's not supported in PHP 5.2. Note that Jetpack [discourages its use.](https://github.com/Automattic/jetpack/blob/master/docs/coding-guidelines.md#versions-supported)